### ISSUE TITLE
apply text edit currently doesn't preserve the buffer fileformat

### ISSUE THREAD

**Issue by prabirshrestha** [prabirshrestha/vim-lsp#659]
LSP spec doesn't care about the new line, Some language servers like typescript lang server returns `\r\n` even though we send them unix line endings.

During textedits we should look at the `&fileformat` and normalize the linendings. In this case if the buffer fileformat is unix we should replace `\r\n` with unix linendings.

```json
{
    "merged": {
      "range": {
        "end": {
          "character": 0,
          "line": 11
        },
        "start": {
          "character": 0,
          "line": 11
        }
      },
      "newText": "import { createLspConnection } from './lsp-connection';\r\n"
    },
    "end_index": 1
  }
```

Else we see `^M` after applying the text edits.

![image](https://user-images.githubusercontent.com/287744/71797013-3fb2bb80-3001-11ea-9820-1ddaac724386.png)

**Comment by hrsh7th** [comment]
I think this issue was solved by #718.
But I have no reproducible environment so I left it at the moment.

**Comment by prabirshrestha** [comment]
This should had been fixed.
